ddtrace/mocktracer/mocktracer.go

```diff
@@ -67,7 +71,26 @@ func (t *mocktracer) StartSpan(operationName string, opts ...ddtrace.StartSpanOp
 	for _, fn := range opts {
 		fn(&cfg)
 	}
-	return newSpan(t, operationName, &cfg)
+	span := newSpan(t, operationName, &cfg)
+
+	t.Lock()
+	defer t.Unlock()
+	if t.openSpans == nil {
+		t.openSpans = make(map[uint64]Span)
+	}
+	t.openSpans[span.SpanID()] = span
```

Comment:
```suggestion
	t.Lock()
	t.openSpans[span.SpanID()] = span
	t.Unlock()
```
Please make the map in `Start`, as I suggested in my previous comment.